src/vos/evtree.c

```diff
@@ -990,7 +1006,8 @@ evt_tcx_create(struct evt_root *root, uint64_t feats, unsigned int order,
 
 	D_ASSERT(root != NULL);
 
-	D_ALLOC_PTR(tcx);
+	d_dtm_restock(vos_tls_get()->vtl_tcx_dtm_type);
```

Comment:
restock() is useful of the critical path, however it's optional.  If you can't call it at a better location then acquire() will call it anyway, so you don't need it here.

Reply:
if we modified the vos API slightly, we could call something like vos_ioc_close(...) to do this after sending the reply.  But we could only do that for the ioc level which is another reason to consider attaching any allocated tree handles to the ioc rather than allocating them internally.